local Axes = require(script.Axes)
export type Axes = Axes.Axes
local BrickColor = require(script.BrickColor)
export type BrickColor = BrickColor.BrickColor
local CFrame = require(script.CFrame)
export type CFrame = CFrame.CFrame
local Color3 = require(script.Color3)
export type Color3 = Color3.Color3
local ColorSequence = require(script.ColorSequence)
export type ColorSequence = ColorSequence.ColorSequence
local ColorSequenceKeypoint = require(script.ColorSequenceKeypoint)
export type ColorSequenceKeypoint = ColorSequenceKeypoint.ColorSequenceKeypoint
local DateTime = require(script.DateTime)
export type DateTime = typeof(DateTime.now())
local Enum = require(script.Enum)
export type Enum = Enum.Enum
local EnumItem = require(script.EnumItem)
export type EnumItem = EnumItem.EnumItem
local Enums = require(script.Enums)
export type Enums = Enums.Enums
local Faces = require(script.Faces)
export type Faces = Faces.Faces
local Font = require(script.Font)
export type Font = Font.Font
local NumberRange = require(script.NumberRange)
export type NumberRange = NumberRange.NumberRange
local NumberSequence = require(script.NumberSequence)
export type NumberSequence = NumberSequence.NumberSequence
local NumberSequenceKeypoint = require(script.NumberSequenceKeypoint)
export type NumberSequenceKeypoint = NumberSequenceKeypoint.NumberSequenceKeypoint
local PhysicalProperties = require(script.PhysicalProperties)
export type PhysicalProperties = PhysicalProperties.PhysicalProperties
local Ray = require(script.Ray)
export type Ray = Ray.Ray
local Rect = require(script.Rect)
export type Rect = Rect.Rect
local UDim = require(script.UDim)
export type UDim = UDim.UDim
local UDim2 = require(script.UDim2)
export type UDim2 = UDim2.UDim2
local Vector2 = require(script.Vector2)
export type Vector2 = Vector2.Vector2
local Vector2int16 = require(script.Vector2int16)
export type Vector2int16 = Vector2int16.Vector2int16
local Vector3 = require(script.Vector3)
export type Vector3 = Vector3.Vector3
local Vector3int16 = require(script.Vector3int16)
export type Vector3int16 = Vector3int16.Vector3int16
local Content = require(script.Content)
export type Content = Content.Content
local UniqueId = require(script:FindFirstChild("UniqueId"))
export type UniqueId = UniqueId.UniqueId

return {
	Axes = Axes,
	BrickColor = BrickColor,
	CFrame = CFrame,
	Color3 = Color3,
	ColorSequence = ColorSequence,
	ColorSequenceKeypoint = ColorSequenceKeypoint,
	DateTime = DateTime,
	Enum = Enum,
	EnumItem = EnumItem,
	Enums = Enums,
	Faces = Faces,
	Font = Font,
	NumberRange = NumberRange,
	NumberSequence = NumberSequence,
	NumberSequenceKeypoint = NumberSequenceKeypoint,
	PhysicalProperties = PhysicalProperties,
	Ray = Ray,
	Rect = Rect,
	UDim = UDim,
	UDim2 = UDim2,
	Vector2 = Vector2,
	Vector2int16 = Vector2int16,
	Vector3 = Vector3,
	Vector3int16 = Vector3int16,
	Content = Content,
	UniqueId = UniqueId,
}
